index.html

```diff
@@ -1733,28 +1733,36 @@ <h3 id="h-mediasource-algorithms" resource="#h-mediasource-algorithms"><span pro
       <section id="mediasource-attach" typeof="bibo:Chapter" resource="#mediasource-attach" property="bibo:hasPart">
         <h4 id="h-mediasource-attach" resource="#h-mediasource-attach"><span property="xhv:role" resource="xhv:heading"><span class="secno">2.4.1 </span>Attaching to a media element</span></h4>
         <p> A <a href="#idl-def-mediasource" class="internalDFN" data-link-type="dfn"><code>MediaSource</code></a> object can be attached to a media element by assigning a <a href="#mediasource-object-url">MediaSource object URL</a> to the media element <code><a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#element-attrdef-media-src">src</a></code> attribute or the src attribute of a &lt;source&gt; inside a media element. A <a href="#mediasource-object-url">MediaSource object URL</a> is created by passing a MediaSource object to <code><a href="#dom-url-createobjecturl">createObjectURL()</a></code>.</p>
-        <p>If the <a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#resource-fetch-algorithm">resource fetch algorithm</a> absolute URL matches the MediaSource object URL, ignore any <code><a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#element-attrdef-media-preload">preload</a></code> attribute of the media element, skip any optional steps for when <code><a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#element-attrdef-media-preload">preload</a></code> equals <code><a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#valdef-htmlmediaelement-preload-none">none</a></code>, and run the following steps right before the <span>"<i>Perform a potentially CORS-enabled fetch</i>"</span> step in the <a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#resource-fetch-algorithm">resource fetch algorithm</a>.</p>
 
+        <p>If the <a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#resource-fetch-algorithm">resource fetch algorithm</a> was invoked with a media provider object that is a MediaSource object or a URL record whose object is a MediaSource object, then let mode be local, skip the first step in the <a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#resource-fetch-algorithm">resource fetch algorithm</a> which may otherwise set mode to remote, and run the following steps at the beginning of the <span>"<i>Otherwise (mode is local)</i>"</span> section of the <a href="https://www.w3.org/TR/html51/semantics-embedded-content.html#resource-fetch-algorithm">resource fetch algorithm</a>.
```

Comment:
This follows the model of the previous section contents, but feels awkward.  The goal is to set the mode to local, skip the first resource fetch algorithm step (which could set mode to remote), add some steps that are MediaSource specific, and modify the "Otherwise (mode is local)" steps to fit.

Since it follows the previous structure here, it's probably okay.  I'd be tempted to say something more like:
`If...then let mode be local, skip the first step in the resource fetch algorithm (which may otherwise set the mode to remote, and add the steps and clarifications below to the "Otherwise (mode is local)" section of the resource fetch algorithm.

This might seem like wordsmithing.  I'm reacting a bit to the "before" aspect of the steps, when they are partly before, and partly alterations of the steps themselves.